packages/lesswrong/components/common/CookieBanner/CookieBanner.tsx

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import { Components, registerComponent } from "../../../lib/vulcan-lib";
+import Button from "@material-ui/core/Button";
+import classNames from "classnames";
+import { useDialog } from "../withDialog";
+import { useCookiePreferences } from "../../hooks/useCookiesWithConsent";
+
+const styles = (theme: ThemeType) => ({
+  bannerContainer: {
+    display: "flex",
+    alignItems: "center",
+    position: "fixed",
+    bottom: 0,
+    left: 0,
+    right: 0,
+    backgroundColor: theme.palette.panelBackground.cookieBanner,
+    zIndex: 1001, // Appear above sunshine sidebar
+    flexDirection: "row",
+    justifyContent: "center",
+    padding: "16px",
+    [theme.breakpoints.down("md")]: {
+      padding: "16px",
+    },
+    [theme.breakpoints.down("xs")]: {
+      padding: 14,
+      flexDirection: "column",
+      display: "grid",
+    },
+  },
+  text: {
+    color: theme.palette.text.invertedBackgroundText,
+    fontWeight: 400,
+    maxWidth: 1200,
+    fontSize: 14,
+    "& a": {
+      textDecoration: "underline",
+      fontWeight: 600,
+      "&:hover": {
+        textDecoration: "underline",
+      },
+    },
+  },
+  buttonGroup: {
+    display: "flex",
+    marginTop: 0,
+    alignItems: "flex-end",
+    marginLeft: 24,
+    [theme.breakpoints.down("xs")]: {
+      marginTop: 12,
+      marginLeft: 0,
+    },
+  },
+  button: {
+    textTransform: "none",
+    color: theme.palette.text.alwaysWhite,
+    whiteSpace: "nowrap",
+    fontSize: 14,
+    marginRight: 6,
+    boxShadow: "none",
+    [theme.breakpoints.up("sm")]: {
+      marginBottom: 0,
+    },
+  },
+  rejectButton: {
+    backgroundColor: theme.palette.grey[600],
+    "&:hover": {
+      backgroundColor: theme.palette.grey[700],
+    },
+  }
+});
+
+const CookieBanner = ({ classes }: { classes: ClassesType }) => {
+  const { openDialog } = useDialog();
+  const { updateCookiePreferences } = useCookiePreferences();
+  
+  const handleAcceptAll = () => {
+    updateCookiePreferences(["necessary", "functional", "analytics"]);
+  }
+  const handleReject = () => {
+    updateCookiePreferences(["necessary"]);
+  }
+
+  const { Typography } = Components;
+  return (
+    <div className={classNames(classes.bannerContainer)}>
+      <Typography variant="body2" className={classes.text}>
+        We use cookies to enhance your experience, by clicking "Accept All" you agree to their use. Customise your{" "}
```

Comment:
Copy suggestions:

We use cookies to enhance your experience. By clicking "Accept all", you agree to their use. Customize your cookie settings for more control, or review our cookie policy here.

Reply:
Wait what's the difference here, just the lowercase "all" and the full stop?

Reply:
Basically, I just wanted the casing to match the button and to have proper sentences (including adding a period at the end).

I feel like this reads awkwardly because it's two sentences smushed together: "We use cookies to enhance your experience, by clicking "Accept all" you agree to their use."